utils/author_catalog.py

```diff
@@ -38,13 +33,31 @@ def parse_data_row(row: list):
     )
 
 
-def is_selected_author(book: Book, language="en"):
+def is_selected_author(book: Book, selected_authors=SELECTED_AUTHORS, language="en"):
+    """
+    Note:
+        'Translator' control should check only for the second element of the strings; otherwise, there is a
+    possibility that it takes the books to have multiple authors, which we are not interested in.
+
+    book.author.split(';')[1] only controls second element for
+    Args:
+        book:
+        selected_authors:
+        language:
+
+    Returns:
+
+    """
+    _book_author = book.author
+    if ";" in book.author and "Translator" in book.author.split(";")[1]:
+        book.author = book.author.split(";")[0]
     if book.author in selected_authors and book.language == language:
+        book.author = _book_author
```

Comment:
kind of didn't like the names. it is confusing that we have two almost the same variable names. could you find better names? most likely the first one is `authors`. also, i think the second `if` should be inside of the first `if`, no?

Reply:
They are the name already. _book_author is the temporary variable. If I don't create this _book_author variable and if get  into line 52 overwrites and turns it into a shorter name format. We don't want that kind of manipulation in the log file. I try to keep original name when we are recording this information to the log file.

No, it shouldn't because not all names contains ";" only entries that have translators contain this character this. So this control was written to catch and transform to the second if-control statement which catch normal cases. If I put the second one into the first one only catch the entry contains Translator.